src/Amazon.CloudWatch.EMF/Model/MetricsContext.cs

```diff
@@ -184,7 +184,10 @@ public List<DimensionSet> GetAllDimensionSets()
         /// <param name="dimensionSets">the dimensionSets to use instead of all existing dimensions and default dimensions.</param>
         public void SetDimensions(params DimensionSet[] dimensionSets)
         {
-            _metricDirective.SetDimensions(dimensionSets.ToList());
+            _metricDirective.SetDimensions(
+                dimensionSets
+                    .Where(dimensionSet => dimensionSet != null)
+                    .ToList());
```

Comment:
This won't fix the ArgumentNullException that you are having when calling `metricsLogger.SetDimensions(null)`.
Your change is checking individual DimensionSet dictionaries instead.

Here is my suggestion:

```suggestion
            if (dimensionSets = null) return;
            _metricDirective.SetDimensions(dimensionSets.ToList());
              
```